projects/Mallard/src/screens/log-in.tsx

```diff
@@ -0,0 +1,384 @@
+                                    buttonStyles={[
+                                        loginStyles.button,
+                                        {
+                                            borderColor: color.primary,
+                                            borderWidth: 1,
+                                            backgroundColor: 'transparent',
+                                        },
+                                    ]}
+                                    textStyles={{ color: color.primary }}
+                                    onPress={() => {
+                                        setHasInputEmail(false)
+                                    }}
+                                >
+                                    Back
+                                </Button>
+                            )}
+                            <Button
+                                center
+                                style={loginStyles.buttonContainer}
+                                buttonStyles={[
+                                    loginStyles.button,
+                                    {
+                                        backgroundColor: color.primary,
+                                    },
+                                ]}
+                                textStyles={{ color: 'white' }}
+                                onPress={() => {
+                                    if (hasInputEmail) {
+                                        if (password.error) {
+                                            setShowError(true)
+                                        } else {
+                                            onSubmit()
+                                        }
+                                    } else {
+                                        if (email.error) {
+                                            setShowError(true)
+                                        } else {
+                                            setHasInputEmail(true)
+                                        }
+                                    }
+                                }}
+                            >
+                                {!hasInputEmail
+                                    ? emailProgressText
+                                    : submitText}
+                            </Button>
+                        </View>
+                        <View style={loginStyles.actionRow}>
+                            {hasInputEmail && (
+                                <Link
+                                    style={loginStyles.resetLink}
+                                    href={resetLink}
+                                >
+                                    Forgot password?
+                                </Link>
+                            )}
+                        </View>
+                    </View>
+                </View>
+            </KeyboardAvoidingView>
+        </View>
+    )
+}
+
+export { Login }
```

Comment:
This file is huge! Any chance of breaking into chunks?

Reply:
Will do this on the next CAS PR - I definitely agree but I think it'll happen as a side effect of some future changes so will just do it there.